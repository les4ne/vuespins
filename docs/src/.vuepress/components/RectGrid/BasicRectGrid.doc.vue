<template>
  <div>
    <RectGrid />
  </div>
</template>

<script>
import { RectGrid } from 'vuespins'

export default {
  name: 'app',
  components: { RectGrid }
}
</script>
