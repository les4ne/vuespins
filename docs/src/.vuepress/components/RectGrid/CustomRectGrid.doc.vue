<template>
  <div>
    <RectGrid color="#fb503b" size="80" speed="5" />
  </div>
</template>

<script>
import { RectGrid } from 'vuespins'

export default {
  name: 'app',
  components: { RectGrid }
}
</script>
