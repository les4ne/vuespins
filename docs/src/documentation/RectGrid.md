# RectGrid

## **Basic**

<br>
<client-only>
<RectGrid />
</client-only>

### Props

Basic "RectGrid" component

| property | propType | required | default | custom |
| :------: | :------: | :------: | :-----: | :----: |
| `color`  |  string  |  false   | #282c34 |   -    |
|  `size`  |  number  |  false   |   50    |   -    |
| `speed`  |  number  |  false   |   1.2   |   -    |

### Usage

<<< @/src/.vuepress/components/RectGrid/BasicRectGrid.doc.vue{3,8,10-13}

## **Custom**

<br>
<client-only>
<RectGrid color="#fb503b" :size="80" :speed="3" />
</client-only>

### Props

Custom "RectGrid" component

| property | propType | required | default | custom  |
| :------: | :------: | :------: | :-----: | :-----: |
| `color`  |  string  |  false   | #282c34 | #fb503b |
|  `size`  |  number  |  false   |   50    |   80    |
| `speed`  |  number  |  false   |   1.2   |    3    |

### Usage

<<< @/src/.vuepress/components/RectGrid/CustomRectGrid.doc.vue{3,8,10-13}
