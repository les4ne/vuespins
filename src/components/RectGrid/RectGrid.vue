<template>
  <div class="vs-rect-grid" :style="cssVars">
    <div class="rect1"></div>
    <div class="rect2"></div>
    <div class="rect3"></div>
    <div class="rect4"></div>
    <div class="rect5"></div>
  </div>
</template>

<script>
export default {
  name: 'vs-rect-grid',
  props: {
    color: {
      type: String,
      default: '#282c34'
    },
    size: {
      type: String,
      default: '50'
    },
    speed: {
      type: String,
      default: '1.2'
    }
  },
  computed: {
    cssVars() {
      return {
        '--vs-rect-grid-size': this.size + 'px',
        '--vs-rect-grid-color': this.color,
        '--vs-rect-grid-speed': this.speed + 's',
        '--vs-rect-grid-delay-1': this.speed - 2.3 + 's',
        '--vs-rect-grid-delay-2': this.speed - 2.2 + 's',
        '--vs-rect-grid-delay-3': this.speed - 2.1 + 's',
        '--vs-rect-grid-delay-4': this.speed - 2 + 's',
        '--vs-rect-grid-delay-5': this.speed - 1.9 + 's'
      }
    }
  }
}
</script>

<style scoped>
.vs-rect-grid {
  width: var(--vs-rect-grid-size);
  height: var(--vs-rect-grid-size);
  text-align: center;
  font-size: 10px;
}

.vs-rect-grid > div {
  background-color: var(--vs-rect-grid-color);
  height: 100%;
  width: 6px;
  display: inline-block;
  -webkit-animation: vs-stretchdelay var(--vs-rect-grid-speed) infinite
    ease-in-out;
  animation: vs-stretchdelay var(--vs-rect-grid-speed) infinite ease-in-out;
}

.vs-rect-grid .rect1 {
  -webkit-animation-delay: var(--vs-rect-grid-delay-1);
  animation-delay: var(--vs-rect-grid-delay-1);
}

.vs-rect-grid .rect2 {
  -webkit-animation-delay: var(--vs-rect-grid-delay-2);
  animation-delay: var(--vs-rect-grid-delay-2);
}

.vs-rect-grid .rect3 {
  -webkit-animation-delay: var(--vs-rect-grid-delay-3);
  animation-delay: var(--vs-rect-grid-delay-3);
}

.vs-rect-grid .rect4 {
  -webkit-animation-delay: var(--vs-rect-grid-delay-4);
  animation-delay: var(--vs-rect-grid-delay-4);
}

.vs-rect-grid .rect5 {
  -webkit-animation-delay: var(--vs-rect-grid-delay-5);
  animation-delay: var(--vs-rect-grid-delay-5);
}

@-webkit-keyframes vs-stretchdelay {
  0%,
  40%,
  100% {
    -webkit-transform: scaleY(0.4);
  }
  20% {
    -webkit-transform: scaleY(1);
  }
}

@keyframes vs-stretchdelay {
  0%,
  40%,
  100% {
    transform: scaleY(0.4);
    -webkit-transform: scaleY(0.4);
  }
  20% {
    transform: scaleY(1);
    -webkit-transform: scaleY(1);
  }
}
</style>
