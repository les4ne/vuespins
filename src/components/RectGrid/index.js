import RectGrid from './RectGrid.vue'

export default RectGrid
