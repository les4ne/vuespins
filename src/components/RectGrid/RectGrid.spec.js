import { shallowMount } from '@vue/test-utils'
import RectGrid from './RectGrid.vue'

describe('RectGrid.vue', () => {
  it('renders correctly', () => {
    expect(shallowMount(RectGrid).element).toMatchSnapshot()
  })
})
