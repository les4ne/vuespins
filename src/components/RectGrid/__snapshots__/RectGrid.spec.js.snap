// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RectGrid.vue renders correctly 1`] = `
<div
  class="vs-rect-grid"
  style="--vs-rect-grid-size: 50px; --vs-rect-grid-color: #282c34; --vs-rect-grid-speed: 1.2s; --vs-rect-grid-delay-1: -1.0999999999999999s; --vs-rect-grid-delay-2: -1.0000000000000002s; --vs-rect-grid-delay-3: -0.9000000000000001s; --vs-rect-grid-delay-4: -0.8s; --vs-rect-grid-delay-5: -0.7s;"
>
  <div
    class="rect1"
  />
   
  <div
    class="rect2"
  />
   
  <div
    class="rect3"
  />
   
  <div
    class="rect4"
  />
   
  <div
    class="rect5"
  />
</div>
`;
